server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -140,32 +167,47 @@ public Query termQuery(Object value, SearchExecutionContext context) {
         }
     }
 
-    private final boolean enabled;
+    private final Mode mode;
```

Comment:
mark this one as `@Nullable`? Might not be immediately clear why it can be null (bwc reasons).